Sources/NIOCore/ByteBuffer-hexdump.swift

```diff
@@ -0,0 +1,188 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+extension ByteBuffer {
+
+    /// Describes a ByteBuffer hexDump format.
+    /// Can be either xxd output compatible, or hexdump compatible.
+    public struct HexDumpFormat {
+        enum Value {
+            case xxdCompatible(maxLength: Int? = nil)
+            case hexDumpCompatible(maxLength: Int? = nil)
+        }
+
+        var value: Value
+        init(_ value: Value) { self.value = value }
+
+        /// A hex dump format compatible with `xxd` CLI utility.
+        public static let xxdCompatible = Self(.xxdCompatible(maxLength: nil))
+
+        /// A hex dump format compatible with `hexdump` command line utility.
+        public static let hexDumpCompatible = Self(.hexDumpCompatible(maxLength: nil))
+
+        /// A hex dump format compatible with `xxd`, clipped to `maxLength` bytes dumped.
+        /// This format will dump first maxLength / 2 bytes, and the last maxLength / 2 bytes, replacing the rest with " ... ".
+        public static func xxdCompatible(maxLength: Int) -> Self {
+            Self(.xxdCompatible(maxLength: maxLength))
+        }
+
+        /// A hex dump format compatible with `hexdump` command line tool.
+        /// This format will dump first maxLength / 2 bytes, and the last maxLength / 2 bytes, with a placeholder in between.
+        public static func hexDumpCompatible(maxLength: Int) -> Self {
```

Comment:
I think we should use `maxBytes` instead of `maxLength`;  you could mistake length for being the length of the output.